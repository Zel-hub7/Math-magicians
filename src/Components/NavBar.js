import { Link } from 'react-router-dom';
import './navBar.css';

const NavBar = () => (
  <nav className="navigation">
    <Link to="./">
      <h1 className="logo">Math Magician</h1>
    </Link>

    <ul className="nav-list">
      <li>
        <Link to="/">Home </Link>
      </li>

      <li>
        <Link to="/Calculator">Calculator</Link>
      </li>

      <li>
        <Link to="/Quotes">Quotes</Link>
      </li>
    </ul>

  </nav>
);
export default NavBar;
