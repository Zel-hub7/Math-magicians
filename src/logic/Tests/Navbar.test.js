import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NavBar from '../../Components/NavBar';

test('renders navigation links correctly', () => {
  render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>,
  );

  expect(screen.queryByText(/Home/i)).toBeTruthy();
  expect(screen.queryByText(/Calculator/i)).toBeTruthy();
  expect(screen.queryByText(/Quotes/i)).toBeTruthy();
});

test('matches snapshot', () => {
  const { container } = render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>,
  );

  expect(container).toMatchSnapshot();
});
