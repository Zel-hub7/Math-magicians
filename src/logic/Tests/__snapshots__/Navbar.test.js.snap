// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <nav
    class="navigation"
  >
    <a
      href="/"
    >
      <h1
        class="logo"
      >
        Math Magician
      </h1>
    </a>
    <ul
      class="nav-list"
    >
      <li>
        <a
          href="/"
        >
          Home 
        </a>
      </li>
      <li>
        <a
          href="/Calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/Quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
